projects/plugins/jetpack/modules/search/class-jetpack-search-customberg.php

```diff
@@ -0,0 +1,145 @@
+	}
+
+	/**
+	 * Prints the dashboard container.
+	 *
+	 * @access public
+	 */
+	public function jetpack_search_admin_page() {
+		// TODO: Spin this function off into a static helper function in a helper class for code reuse.
+		$protocol   = is_ssl() ? 'https' : 'http';
+		$static_url = apply_filters( 'jetpack_static_url', "{$protocol}://en.wordpress.com/i/loading/loading-64.gif" );
+		?>
+			<div id="jp-search-customization" class="jp-search-customization-dashboard">
+				<div class="hide-if-no-js"><img class="jp-search-loader" width="32" height="32" alt="<?php esc_attr_e( 'Loading&hellip;', 'jetpack' ); ?>" src="<?php echo esc_url( $static_url ); ?>" /></div>
+				<div class="hide-if-js"><?php esc_html_e( 'Your Search customization page requires JavaScript to function properly.', 'jetpack' ); ?><div />
+			</div>
+		<?php
+	}
+
+	/**
+	 * Enqueue admin styles.
+	 */
+	public function load_admin_styles() {
+		\Jetpack_Admin_Page::load_wrapper_styles();
+
+		wp_enqueue_style(
+			'jp-search-customize',
+			plugins_url( '_inc/build/instant-search/jp-search-configure-main.bundle.css', JETPACK__PLUGIN_FILE ),
+			array(
+				'wp-components',
+				'wp-block-editor',
+			),
+			JETPACK__VERSION
+		);
+	}
+
+	/**
+	 * Enqueue admin scripts.
+	 */
+	public function load_admin_scripts() {
+		$script_deps_path    = JETPACK__PLUGIN_DIR . '_inc/build/instant-search/jp-search-configure-main.bundle.asset.php';
+		$script_dependencies = array( 'react', 'wp-i18n', 'wp-polyfill' );
+		if ( file_exists( $script_deps_path ) ) {
+			$asset_manifest      = include $script_deps_path;
+			$script_dependencies = $asset_manifest['dependencies'];
+		}
+
+		wp_enqueue_script( 'jp-search-customization-tracks', '//stats.wp.com/w.js', array(), gmdate( 'YW' ), true );
+
+		wp_enqueue_script(
+			'jp-search-customization',
+			plugins_url( '_inc/build/instant-search/jp-search-configure-main.bundle.js', JETPACK__PLUGIN_FILE ),
+			$script_dependencies,
+			JETPACK__VERSION,
+			true
+		);
+
+		// Use wp_add_inline_script instead of wp_localize_script, see https://core.trac.wordpress.org/ticket/25280.
+		wp_add_inline_script( 'jp-search-customization', 'var JetpackInstantSearchOptions=JSON.parse(decodeURIComponent("' . rawurlencode( wp_json_encode( Jetpack_Search_Helpers::generate_initial_javascript_state() ) ) . '"));', 'before' );
+		wp_add_inline_script(
+			'jp-search-customization',
+			"window.jetpackSearchCustomizeInit( 'jp-search-customization' )"
+		);
+
+		// Inject valid post type definitions.
+		wp_add_inline_script( 'jp-search-customization', 'var JetpackInstantSearchValidPostTypes=JSON.parse(decodeURIComponent("' . rawurlencode( wp_json_encode( get_post_types( array( 'exclude_from_search' => false ), 'objects' ) ) ) . '"));', 'before' );
+	}
+
+	/**
+	 * Determine if the requisite page should be added to wp-admin.
+	 *
+	 * @return boolean
+	 */
+	private function should_add_page() {
+		return Jetpack_Plan::supports( 'search' );
```

Comment:
Should we handle non-connected / sites in offline mode here, as well as limiting it to site admins?

Reply:
Good call, adding the checks now.

Reply:
Quick update: I realized we already check for site admins [when we add the customization page](https://github.com/Automattic/jetpack/pull/20282/files/#diff-119cb8cdd124378b37ca4e8f9df2a855d0208f56e0bb47a986626b61698a7b95R63), so I removed a redundant check from here.